File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/generic_capsule/data/dto/CapsuleCreateRequestDto.java
package site.timecapsulearchive.core.domain.capsule.generic_capsule.data.dto;

import java.time.ZonedDateTime;
import java.util.List;
import lombok.Builder;
import org.locationtech.jts.geom.Point;
import site.timecapsulearchive.core.domain.capsule.entity.Capsule;
import site.timecapsulearchive.core.domain.capsule.entity.CapsuleType;
import site.timecapsulearchive.core.domain.capsuleskin.entity.CapsuleSkin;
import site.timecapsulearchive.core.domain.member.entity.Member;
import site.timecapsulearchive.core.infra.map.data.dto.AddressData;

@Builder
public record CapsuleCreateRequestDto(
    Long capsuleSkinId,
    String title,
    String content,
    Double longitude,
    Double latitude,
    AddressData addressData,
    ZonedDateTime dueDate,
    List<String> imageNames,
    List<String> videoNames,
    String directory
) {

    public Capsule toCapsule(
        final Point point,
        final Member member,
        final CapsuleSkin capsuleSkin,
        final CapsuleType capsuleType
    ) {
        return Capsule.builder()
            .title(title)
            .content(content)
            .point(point)
            .address(addressData.toAddress())
            .type(capsuleType)
            .member(member)
            .dueDate(dueDate)
            .capsuleSkin(capsuleSkin)
            .build();
    }
}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/common/CapsuleCreateRequestDto.kt
package com.droidblossom.archive.data.dto.common

import com.droidblossom.archive.data.dto.capsule.response.AddressDataDto

data class CapsuleCreateRequestDto(
    val capsuleSkinId: Long,
    val content: String,
    val directory: String,
    val dueDate: String?,
    val imageNames: List<String>?,
    val videoNames: List<String>?,
    val addressData: AddressDataDto,
    val latitude: Double,
    val longitude: Double,
    val title: String
)